src/sign.js

```diff
@@ -1,94 +1,155 @@
-  if (!target.value) {
-    createErrMsg(password, "비밀번호를 입력해 주세요.");
-    passwordInput.classList.add("err");
+  const password = target.value;
+  if (!password) {
+    invalidPasswordMsg.classList.remove("errNone");
+    return;
   } else {
-    removeErrMsg(password);
-    passwordInput.classList.remove("err");
+    invalidPasswordMsg.classList.add("errNone");
+    isPasswordValid = true;
   }
 };
 
-// 로그인 버튼 클릭 시 이벤트 함수
-const submitLogin = () => {
+/**
+ * 주어진 경로로 POST 요청을 보내는 함수입니다.
+ * @param {string} path - 요청을 보낼 경로
+ * @param {Object} data - 요청에 포함될 데이터
+ * @param {string} authorized - 인증 토큰 (선택적)
+ * @returns {Promise<Object>} - 응답 데이터를 담은 Promise 객체
+ */
+const postRequest = async (path, data, authorized) => {
+  const res = await fetch(`https://bootcamp-api.codeit.kr/${path}`, {
+    method: "POST",
+    headers: {
+      "Access-Control-Allow-Origin": "*",
+      "Content-Type": "application/json",
+      Autorization: authorized ? `Bearer ${authorized}` : "",
+    },
+    body: JSON.stringify(data),
+  });
+  return await res.json();
+};
+
+/**
+ * 로그인을 제출하는 함수입니다.
+ * @returns {void}
+ */
+const submitLogin = async () => {
   const emailValue = document.querySelector("#email").value;
   const passwordValue = document.querySelector("#password").value;
-  if (emailValue === TEST_EMAIL && passwordValue === TEST_PASSWORD) {
-    location.href = "/folder.html";
+  if (isEmailValid && isPasswordValid) {
+    const res = await postRequest("api/sign-in", { email: emailValue, password: passwordValue }, null);
+    window.localStorage.setItem("accessToken", res.accessToken);
+    window.location.href = "folder.html";
   } else {
-    createErrMsg(email, "이메일을 확인해 주세요.");
-    createErrMsg(password, "비밀번호를 확인해 주세요.");
+    emailValidation({ target: emailInput });
+    passwordValidator({ target: passwordInput });
   }
 };
 
-// 비밀번호 눈 아이콘 클릭 시 이벤트 함수
-const handleToPasswordEye = () => {
-  const passwordType = passwordInput.getAttribute("type");
+/**
+ * 비밀번호의 눈 아이콘을 클릭했을 때, 비밀번호를 보여주거나 가리는 함수입니다.
+ * @param {string} type - 비밀번호의 타입입니다. 현재는 PASSWORD만 가능합니다.
+ * @returns {void}
+ */
+const handleToPasswordEye = (type) => {
+  const passwordType = PASSWORD_CONFIG[type];
+  const passwordAttribute = PASSWORD_CONFIG[type].INPUT.getAttribute("type");
 
-  passwordType === "password"
-    ? (passwordInput.setAttribute("type", "text"), eyeOff.classList.toggle("none"), eyeOn.classList.toggle("none"))
-    : (passwordInput.setAttribute("type", "password"), eyeOff.classList.toggle("none"), eyeOn.classList.toggle("none"));
+  if (passwordAttribute === "password") {
+    passwordType.INPUT.setAttribute("type", "text");
+    passwordType.EYE_OFF.classList.toggle("none");
+    passwordType.EYE_ON.classList.toggle("none");
+    return;
+  } else {
+    passwordType.INPUT.setAttribute("type", "password");
+    passwordType.EYE_OFF.classList.toggle("none");
+    passwordType.EYE_ON.classList.toggle("none");
+    return;
+  }
```

Comment:
조건 문 내부의 코드가 굉장히 유사한 것 같습니다.
아래와 같이 수정해보는건 어떨까요?

```suggestion
    const isPasswordVisible = passwordAttribute !== "password";
    passwordType.INPUT.setAttribute("type",  isPasswordVisible ? "password" : "text");
    passwordType.EYE_OFF.classList.toggle("none");
    passwordType.EYE_ON.classList.toggle("none");
```

Reply:
코드에서 다른 부분은 type을 셋업하는 부분 뿐이기 때문에 위와 같이 특정 변수로 삼항 연산자를 활용해 분기를 수행할 수 있습니다.